course6/toyapp-adam/math/vector.cc

```diff
@@ -0,0 +1,49 @@
+#include "vector.h"
+
+namespace Math {
+
+Vector3D Vector3D::operator-() const
+{
+	return {-x, -y, -z};
+}
+
+Vector3D Vector3D::operator+(const Vector3D& rhs) const
+{
+	return {x + rhs.x, y + rhs.y, z + rhs.z};
+}
+
+Vector3D Vector3D::operator-(const Vector3D& rhs) const
+{
+	return {x - rhs.x, y - rhs.y, z - rhs.z};
+}
+
+Vector3D Vector3D::operator*(float scalar) const
```

Comment:
For performance and readability reasons, it's better to keep the one-lines in the class definition.